artifacts-guidance.md

```diff
@@ -1,5 +1,5 @@
 # Guidance for Artifacts Authors
 
 Content other than OCI container images MAY be packaged using the image manifest.
-When this is done, the `config.mediaType` value MUST be set to a value specific to the artifact type or the [empty value](manifest.md#guidance-for-an-empty-descriptor).
-Additional details and examples are provided in the [image manifest specification](manifest.md#guidelines-for-artifact-usage).
+When this is done, the `config.mediaType` value will not be a known OCI image config [media type](media-types.md).
```

Comment:
Hmm, isn't it totally valid for a non-image artifact to set `artifactType` but still use an OCI media type for config? (for example, a fully compliant image config)

I don't see an obvious use case for doing so, but this feels like an unnecessary restriction to me.

Reply:
In that scenario, how would a container runtime differentiate between a container image and something the runtime should ignore?

Reply:
I believe what buildkit is doing would politely be referred to as a hack or kludge to bypass registry filters. There's a [PR to fix it](https://github.com/moby/buildkit/pull/3610) but it's blocked because at least one registry refuses unknown/non-image content. The [config spec](https://github.com/opencontainers/image-spec/blob/main/config.md) describes it as:

> This specification outlines the JSON format describing images for use with a container runtime and execution tool and its relationship to filesystem changesets, described in [Layers](https://github.com/opencontainers/image-spec/blob/main/layer.md).

I hesitate to recommend other uses that may cause issues for existing runtimes that don't know about the `artifactType` field.

Reply:
From today's call, we added a note to clarify that some existing artifacts did not follow this OCI advice, and softened the "will" to a "should" even though this part is guidance and not treated the same as the other parts of the spec.

Reply:
Just to clarify, I wasn't using BuildKit's objects as an example of "a good artifact", but rather as an example object that does currently exist today, and which runtimes already have to deal with in some way (and I think already probably deal with in the only way they reasonably can, by balking at the layer media types).

(I also suggested `artifactType` to them in the PR you linked :smile:)